src/Lykke.Job.CandlesProducer/Modules/JobModule.cs

```diff
@@ -138,10 +138,11 @@ private void RegisterCandlesServices(ContainerBuilder builder)
                 .As<ISnapshotSerializer>();
 
             builder.RegisterType<CandlesGeneratorSnapshotRepository>()
-                .As<ISnapshotRepository<ImmutableDictionary<string, ImmutableList<ICandle>>>>()
-                .WithParameter(TypedParameter.From(AzureBlobStorage.Create(snapshotsConnStringManager, maxExecutionTimeout: TimeSpan.FromMinutes(5))));
+                .As<ISnapshotRepository<ImmutableDictionary<string, ICandle>>>()
+                .WithParameter(TypedParameter.From(AzureBlobStorage.Create(snapshotsConnStringManager, maxExecutionTimeout: TimeSpan.FromMinutes(5))))
+                .SingleInstance();
 
-            builder.RegisterType<SnapshotSerializer<ImmutableDictionary<string, ImmutableList<ICandle>>>>()
+            builder.RegisterType<SnapshotSerializer<ImmutableDictionary<string, ICandle>>>()
```

Comment:
Can this be singleton?

Reply:
Why it should be singleton? Construction of this object is very cheap and stateless

Reply:
I did not say "should" just "can". No ctor in any case chapter then any ctor + less garbage for GC.

Reply:
But from the other hand, singleton will increase number of the long lived objects in gen2, which worse for GC. Allocate and collect an simple object in the gen0 - is very easy for GC. Of course, this is discussable question, but according to my expirience, better to avoid long lived objects until they construction is expensive or some state should be stored along app lifetime

Reply:
Can't compose a text longer than yours ) Agree. That was just a suggestion.

Reply:
Remark starting such a discussion valuable, indeed